cdap-ui/app/features/hydrator/services/create/stores/config-store.js

```diff
@@ -234,6 +232,14 @@ class ConfigStore {
     }
     this.emitChange();
   }
+  setEngine(engine) {
+    if (this.state.config.artifact === this.GLOBALS.etlBatch) {
+      this.state.config.engine = engine || 'mapreduce';
+    }
+  }
+  getEngine() {
+    return this.state.config.engine || 'mapreduce';
```

Comment:
So right now only batch pipelines can have engines. I can add a check here but might not be useful as we are using the engine only for a batch pipeline in detailed view.